activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -78,7 +78,7 @@ def initialize(constructor = {})
     def default(*args)
       arg_key = args.first
 
-      if include?(key = convert_key(arg_key))
+      if include?(key = convert_key(arg_key)) && args.size > 0
         self[key]
```

Comment:
Huh. This line (i.e., not your change) sounds notably unlike Hash#default. :confused:

```
[2] pry(main)> Hash.new { 5 }.merge(1 => 2).default(3)
=> 5
[3] pry(main)> Hash.new { 5 }.merge(1 => 2).with_indifferent_access.default(3)
=> 5
[4] pry(main)> Hash.new { 5 }.merge(1 => 2).default(1)
=> 5
[5] pry(main)> Hash.new { 5 }.merge(1 => 2).with_indifferent_access.default(1)
=> 2
```